code/api/mobi-api/Repository/ProductRepository.cs

```diff
@@ -1,24 +1,36 @@
 ﻿using mobi_api.Model;
+using mobi_api.DAO;
+using mobi_api.Services;
 using System.Text.Json;
 
+
 namespace mobi_api.Repository
 {
     public interface IProductRepository
     {
         List<Product> GetAllProducts();
+        List<ProductEntity> GetProductByStoreId(Guid StoreId);
     }
     public class ProductsRepository : IProductRepository
     {
-        public ProductsRepository()
+        private readonly MobiConsumerContext _dbContext;
+        public ProductsRepository(MobiConsumerContext mobiConsumerContext)
         {
-
+            _dbContext = mobiConsumerContext;
         }
 
         public List<Product> GetAllProducts()
         {
             return this.GetMockProductFromJson();
         }
 
+        public List<ProductEntity> GetProductByStoreId(Guid StoreId)
+        {
+            var store = _dbContext.Stores.Find(StoreId);
```

Comment:
change to
_dbContext.Products.Where(product => product.StoreId.equas(StoreId)).toList();